src/dbt_client/dbtCloudIntegration.ts

```diff
@@ -395,6 +412,8 @@ export class DBTCloudProjectIntegration
   }
 
   async unsafeCompileQuery(query: string): Promise<string | undefined> {
+    this.telemetry.sendTelemetryEvent("dbtCloud:unsafeCompileQuery");
+    this.throwIfNotAuthenticated();
```

Comment:
I think this should work differently. we should send an event in dbtProject and add a projectIntegration dimension. that way we can easily compare events for both cloud vs core.